pkg/licensing/license_manager.go

```diff
@@ -0,0 +1,100 @@
+package licensing
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
+	"github.com/bacalhau-project/bacalhau/pkg/lib/license"
+)
+
+// LicenseManager handles license management and validation for the node
+type LicenseManager struct {
+	bacalhauConfig    *types.Bacalhau
+	licenseConfigured bool
+	licenseToken      string
+	licenseValidator  *license.LicenseValidator
+}
+
+// NewLicenseManager creates and initializes a new LicenseManager
+func NewLicenseManager(config *types.Bacalhau) (*LicenseManager, error) {
+	if config == nil {
+		return nil, errors.New("config cannot be nil")
+	}
+
+	validator, err := license.NewOfflineLicenseValidator()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create license validator: %w", err)
+	}
+
+	licensePath := config.Orchestrator.License.LocalPath
+	licenseConfigured := licensePath != ""
+
+	var licenseToken string
+	if licenseConfigured {
+		// Try to read the license file, fail if the file is not found or malformed
+		// Will not inspect the license itself.
+		licenseData, err := os.ReadFile(licensePath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read license file: %w", err)
+		}
+
+		// Verify the JSON structure
+		var licenseFile struct {
+			License string `json:"license"`
+		}
+		if err := json.Unmarshal(licenseData, &licenseFile); err != nil {
+			return nil, fmt.Errorf("failed to parse license file: %w", err)
+		}
+		licenseToken = licenseFile.License
+	}
+
+	return &LicenseManager{
+		bacalhauConfig:    config,
+		licenseConfigured: licenseConfigured,
+		licenseToken:      licenseToken,
+		licenseValidator:  validator,
+	}, nil
+}
+
+// ValidateLicense validates the current license token and returns the license claims
+func (l *LicenseManager) ValidateLicense() (*license.LicenseClaims, error) {
```

Comment:
The validation should happen in the constructor. This method should be more like get claims or something
 similar

Reply:
@wdbaruni License validation is time dependent, i.e. the license has an expiry date. If we called validate in the constructor once and only return the claims after that, then the license will be valid for as long as the instance is alive. Currently we instantiate a license manager instance when. node started. It will read the license file once, add the token to memory, a whenever `ValidateLicense` or `ValidateLicenseWithNodeCount` are called, it will check the license signature and expiry date. 

Having the the validation in the constructor will defeat that purpose, unless we are instantiating the manager each time we want to validate the license and we need to reading that license file each time.

Reply:
I see. I think the method names are a little confusing. As a user of the license manager, I would expect to have these methods and behaviours:
- `License` which will get all the license information, even if it was expired. For example callers to `bacalhau agent license` should get a license, see when it has expired, and what claims it had. We shouldn't just fail
- `Expired` return true if the license has expired. Or maybe just `IsValid` and return true if the license is still valid
- `Claims` return the claims, and empty claims if expired
- `NodeCount` returns the number of nodes it supports. You can delay this until we implement orchestrator warnings